docs/contributing/continuing-unfinished-work.md

```diff
@@ -0,0 +1,83 @@
+# Continuing unfinished work
+
+Sometimes, work on an issue is started, but not brought to completion. This may
+happen for a variety of reasons — the contributor working on the project gets
+busy, maintainers cannot prioritize reviewing the work, a contributor doesn't
+have the skills required to complete the project, there is
+an unexpected technical challenge or blocker, etc.
+
+Completing work that someone else has started is a great way to contribute! Here
+are the steps required:
+
+1. [Find work to be completed.](#find-work-to-be-completed)
+1. [Review existing work and feedback.](#review-existing-work-and-feedback)
+1. [Decide how to use prior work.](#decide-how-to-use-prior-work)
+1. [Present your pull request.](#present-your-pull-request)
+
+## Find work to be completed
+
+In Zulip's server and web app [repository](https://github.com/zulip/zulip), pull
+requests that have significant work towards something valuable are often tagged
+with a [completion candidate label][completion-candidate] label. You can review
+this label for issues that you find interesting and have the skills to complete.
+
+In general, it's common to see one or more pull request linked to an issue
+you're interested in.
+
+## Review existing work and feedback
+
+Any time there are pull requests linked to the issue you are working on, start
+by reviewing the existing work. Read the code, and pay close attention to any
+feedback the pull request received. This will help you avoid any pitfalls other
+contributors encountered.
+
+## Decide how to use prior work
+
+Consider how to use prior work on the issue. In your best judgment, is the
+proposed solution on the right track? If there's reviewer feedback, it should
+help you figure this out.
+
+If prior work looks like a good start:
+
+1. Pull down the existing pull request.
+1. Rebase it on the current version of the `main` branch.
+1. Carefully address any open feedback from reviewers.
+1. Make any other changes you think are needed, including completing any parts
+   of the work that had not been finished.
```

Comment:
I think we need a step that's something like "self-review, test, and revise the work, including potentially splitting out preparatory commits to make it easier to read, until you're proud of the resulting series of commits and are prepared to argue that it is the best work you can produce to complete the issue".

Reply:
Split into two sentences and added.